Microsoft.Toolkit.Uwp.UI.Controls/QuickReturnHeader/QuickReturnHeader.cs

```diff
@@ -0,0 +1,186 @@
+﻿// ******************************************************************
+// Copyright (c) Microsoft. All rights reserved.
+// This code is licensed under the MIT License (MIT).
+// THE CODE IS PROVIDED “AS IS”, WITHOUT WARRANTY OF ANY KIND, EXPRESS OR IMPLIED,
+// INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT.
+// IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM,
+// DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT,
+// TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION WITH
+// THE CODE OR THE USE OR OTHER DEALINGS IN THE CODE.
+// ******************************************************************
+using System;
+using Windows.UI.Composition;
+using Windows.UI.Xaml;
+using Windows.UI.Xaml.Controls;
+using Windows.UI.Xaml.Hosting;
+using Windows.UI.Xaml.Media;
+
+namespace Microsoft.Toolkit.Uwp.UI.Controls
+{
+    /// <summary>
+    /// Quick return header control to be used with ListViews
+    /// </summary>
+    public class QuickReturnHeader : ContentControl
+    {
+        public QuickReturnHeader()
+        {
+            HorizontalContentAlignment = HorizontalAlignment.Stretch;
+        }
+
+        /// <summary>
+        /// Identifies the <see cref="IsQuickReturnEnabled"/> property.
+        /// </summary>
+        public static readonly DependencyProperty IsQuickReturnEnabledProperty =
+            DependencyProperty.Register("IsQuickReturnEnabled", typeof(bool), typeof(QuickReturnHeader), new PropertyMetadata(true, OnIsQuickReturnEnabledChanged));
+
+        /// <summary>
+        /// Gets or sets a value indicating whether the quick return header is enabled.
+        /// If true the quick return behavior is used.
+        /// If false regular header behavior is used.
+        /// Default is true.
+        /// </summary>
+        public bool IsQuickReturnEnabled
+        {
+            get { return (bool)GetValue(IsQuickReturnEnabledProperty); }
+            set { SetValue(IsQuickReturnEnabledProperty, value); }
+        }
+
+        /// <summary>
+        /// Gets or sets the ListView this header belongs to
+        /// </summary>
+        public ListView TargetListView { get; set; }
+
+        protected override void OnApplyTemplate()
+        {
+            if (TargetListView != null)
+            {
+                scrollViewer = GetScrollViewer(TargetListView);
+
+                // Place items below header
+                var panel = TargetListView.ItemsPanelRoot;
+                Canvas.SetZIndex(panel, -1);
+            }
+
+            if (scrollViewer != null)
+            {
+                scrollViewer.ViewChanged += (sender, args) =>
```

Comment:
Ideally, these events should be unsubscribed from first before adding a subscription, just in case the template gets called multiple times.